### PR TITLE
Remove feature gates for flushmode and batchmanager

### LINKED ISSUE #7365: Remove batchmanager and default FlushMode related feature gates
The following 2 PRs:
- https://github.com/microsoft/FluidFramework/pull/7071 (related https://github.com/microsoft/FluidFramework/issues/4048)
- https://github.com/microsoft/FluidFramework/pull/7178 (related https://github.com/microsoft/FluidFramework/issues/3788)

are shipping 2 related features behind a feature gate.

We need to ensure that enabling the feature gate does not break existing clients, provide guidance about how to use the APIs to go around any issues and to ultimately remove the feature-gate and the old code. This issue tracks that work.

Acceptance criteria:
- figure out what/where breaks by enabling the feature gate
- fix/follow-up fixes for eventual issues found
- remove feature gate and unused code (such as the BatchManger)

### PR DESCRIPTION
**Pull request by andre4i**
closes https://github.com/microsoft/FluidFramework/issues/7365

This will be reviewed/merged after the respective experiments are concluded.

### CODE REVIEW

File: packages/test/test-utils/src/retry.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    defaultValue: T,
+    maxTries: number,
+    currentTry: number,
+    backOffMs: number,
+): Promise<T> => {
+    if (currentTry >= maxTries) {
+        return Promise.resolve(defaultValue);
+    }
+
+    await delay(currentTry * backOffMs);
```
[the remark this answers is not in the document]
Comment (andre4i) [reply packages/test/test-utils/src/retry.ts]: This is a test utility currently only used for the puppeteer driven tests. I've changed it, who knows where someone might use it and then need to debug it.

File: examples/data-objects/todo/tests/todo.test.ts [hunk 2]
```diff
@@ -4,8 +4,23 @@
  */
 
 import { globals } from "../jest.config";
+import { retryWithEventualValue } from "@fluidframework/test-utils";
 
 describe("ToDo", () => {
+    const getItemUrl = async (index: number) =>
```
Comment (vladsud) [inline examples/data-objects/todo/tests/todo.test.ts:10]: It feels like changes like that can and should go in as a separate PR, before you make any other changes.

File: packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts [hunk 3]
```diff
@@ -223,7 +223,8 @@ describeNoCompat("stashed ops", (getTestObjectProvider) => {
         assert.strictEqual(map2.get(testKey), bigString);
     });
 
-    it("doesn't resend successful chunked op", async function() {
+    // TODO: investigate reenabling this test if still relevant without a BatchManager
```
Comment (vladsud) [inline packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:226]: reference to issue is better than TODO comment.

**Review by vladsud — APPROVED** [review]
I glanced through it briefly. Looks reasonable, but I'd break it into smaller chunks

File: packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts (hunk 3 above)
Comment (andre4i) [reply packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:226]: I agree. The reason I left it as todo is to finish the investigation BEFORE making this PR ready for review. This test also has some issues with t9s.

**Comment by andre4i** [comment]
Actually, `packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts@"doesn't resend successful chunked op"` may be a legitimate concern. The behavior of the test is cryptic, as it fails because the `ensuresynchronized` call times out. But the reason for that is that the container does not fire the `saved` event -> because `replayPendingStates` cannot send ops -> because the container is not connected -> because it closed with `0x16f`

File: packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts (hunk 3 above)
Comment (andre4i) [reply packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:226]: As long as we don't do chunking, this is irrelevant.

**Comment by andre4i** [comment]
The feature gates are enabled and have been for the past few weeks. Production rollout is at 100%, so this code is safe to remove. I will do so before the release.

File: packages/drivers/local-driver/src/localDocumentDeltaConnection.ts [hunk 4]
```diff
@@ -63,12 +63,7 @@ export class LocalDocumentDeltaConnection extends DocumentDeltaConnection {
     }
 
     protected submitCore(type: string, messages: IDocumentMessage[]) {
-        if (this.isBatchManagerDisabled) {
-            this.emitMessages(type, [messages]);
-        } else {
-            this.submitManager.add(type, messages);
-            this.submitManager.drain();
-        }
+        this.emitMessages(type, [messages]);
```
Comment (vladsud) [inline packages/drivers/local-driver/src/localDocumentDeltaConnection.ts:66]: Can you please open a follow-up bug to rething protocol? There is no need to have array of array ops any more, I believe. That said, possibly we want to delay any such work until we get rid of notion of batches in protocol altogether, and just say - it's an array of loosely connected messages :)

File: packages/loader/container-loader/src/deltaManager.ts [hunk 5]
```diff
@@ -575,9 +576,13 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
             // DeltaManager overall policy - drop all ops on disconnection and rely on
             // container runtime to deal with resubmitting any ops that did not make it through.
             // So drop them, but also raise error event to look into details.
-            this.logger.sendErrorEvent({
+            this.logger.sendTelemetryEvent({
```
Comment (vladsud) [inline packages/loader/container-loader/src/deltaManager.ts:579]: I do not think there is a value in this event at all. I.e. it's either an error (i.e. we believe it should not happen), or it's normal flow that is handled, covered by UT, but really does not matter in terms of logging.
Also reason is not package data - it comes from socket (in majority cases) and we freely log it already.

**Review by vladsud — APPROVED** [review]

File: packages/drivers/local-driver/src/localDocumentDeltaConnection.ts (hunk 4 above)
Comment (andre4i) [reply packages/drivers/local-driver/src/localDocumentDeltaConnection.ts:66]: https://github.com/microsoft/FluidFramework/issues/9322
